core/src/main/scala/org/http4s/ResponseCookie.scala

```diff
@@ -117,4 +120,175 @@ final case class ResponseCookie(
 
   def clearCookie: headers.`Set-Cookie` =
     headers.`Set-Cookie`(copy(content = "", expires = Some(HttpDate.Epoch), maxAge = Some(0L)))
+
+  private def withExpires(expires: HttpDate): ResponseCookie =
+    copy(expires = Some(expires))
+
+  private def withMaxAge(maxAge: Long): ResponseCookie =
+    copy(maxAge = Some(maxAge))
+
+  private def withDomain(domain: String): ResponseCookie =
+    copy(domain = Some(domain))
+
+  private def withPath(path: String): ResponseCookie =
+    copy(path = Some(path))
+
+  private def withSecure(secure: Boolean): ResponseCookie =
+    copy(secure = secure)
+
+  private def withHttpOnly(httpOnly: Boolean): ResponseCookie =
+    copy(httpOnly = httpOnly)
+
+  private def withSameSite(sameSite: SameSite): ResponseCookie =
+    copy(sameSite = sameSite)
+
+  private def addExtension(extension: String): ResponseCookie =
+    copy(extension = self.extension.fold(Option(extension))(old => Some(old + "; " + extension)))
+}
+
+object ResponseCookie {
+  private[http4s] val parser: Parser1[ResponseCookie] = {
+    import Parser.{char, charIn, failWith, ignoreCase1, pure, rep, string1}
+    import Rfc2616.Rfc1123Date
+    import Rfc5234.{digit, dquote}
+
+    /* token             = <token, defined in [RFC2616], Section 2.2> */
+    def token = Rfc2616.token
+
+    /* cookie-name       = token */
+    def cookieName = token
+
+    /* cookie-octet      = %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
+     *                   ; US-ASCII characters excluding CTLs,
+     *                   ; whitespace DQUOTE, comma, semicolon,
+     *                   ; and backslash
+     */
+    val cookieOctet = charIn(
+      Set(0x21.toChar) ++
+        Set(0x23.toChar to 0x2b.toChar: _*) ++
+        Set(0x2d.toChar to 0x3a.toChar: _*) ++
+        Set(0x3c.toChar to 0x5b.toChar: _*) ++
+        Set(0x5d.toChar to 0x7e.toChar: _*))
+
+    /* cookie-value      = *cookie-octet / ( DQUOTE *cookie-octet DQUOTE ) */
+    val cookieValue = rep(cookieOctet).orElse(dquote *> rep(cookieOctet) <* dquote).string
```

Comment:
I think the lack of semicolon in the dquoted part is exactly where people complain that we're too strict?